src/VisualStudio/Core/Def/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -118,6 +118,9 @@ internal sealed class SourceGeneratedFileManager : IOpenTextBufferEventListener
                 File.WriteAllText(temporaryFilePath, "");
             }
 
+            // Normalize file path (hint name can contain mix of slashes).
+            temporaryFilePath = Path.GetFullPath(temporaryFilePath);
```

Comment:
can this throw an IO exception?

Reply:
Not IOException, but it can throw one of the following. The calls above can throw those as well, though.

```cs
// Exceptions:
//   T:System.ArgumentException:
//     path is a zero-length string, contains only white space, or contains one or more
//     of the invalid characters defined in System.IO.Path.GetInvalidPathChars. -or-
//     The system could not retrieve the absolute path.
//
//   T:System.Security.SecurityException:
//     The caller does not have the required permissions.
//
//   T:System.ArgumentNullException:
//     path is null.
//
//   T:System.NotSupportedException:
//     path contains a colon (":") that is not part of a volume identifier (for example,
//     "c:\").
//
//   T:System.IO.PathTooLongException:
//     The specified path, file name, or both exceed the system-defined maximum length.
```

Reply:
Anyway, removed this line.